cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -121,12 +121,22 @@ private Integer runningProgramCount(Id.Program program, String runId) throws Exc
     return null;
   }
 
+  private void waitForStatusCode(final String path, final Id.Program program, final int expectedStatusCode)
+    throws Exception {
+    Tasks.waitFor(true, new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        HttpResponse response = doPost(getVersionedAPIPath(path, Constants.Gateway.API_VERSION_3_TOKEN,
+                                                           program.getNamespaceId()));
+        return expectedStatusCode == response.getStatusLine().getStatusCode();
+      }
+    }, 180, TimeUnit.SECONDS);
```

Comment:
Let's reduce it then. If there's something wrong in the environment, it will unnecessarily keep waiting for 3 min otherwise. How about 10 seconds?